src/deepsparse/v2/text_generation/process_outputs.py

```diff
@@ -52,19 +51,20 @@ def _create_generated_text_output(
         )
 
     def run(
-        self, inp: CompileGenerationsOutput, inference_state: InferenceState, **kwargs
+        self,
+        generated_tokens: numpy.ndarray,
+        generated_logits: numpy.ndarray,
+        finished_reason: list,
+        inference_state: InferenceState,
+        **kwargs,
     ):
         generation_config = inference_state.current_state.get("generation_config")
-        generated_tokens = inp.generated_tokens
-        generated_logits = (
-            inp.generated_logits if generation_config.output_scores else None
-        )
-        finished_reason = inp.finished_reason
+        generated_logits = generated_logits if generation_config.output_scores else None
         sequences = self.tokenizer.batch_decode(
             generated_tokens, skip_special_tokens=True
         )
 
-        finished_reason = [f for f in finished_reason if f]
+        finished_reason = [f[-1] for f in finished_reason]
```

Comment:
What if `f[-1]`?

Reply:
we produce a finish reason for every generated token but it is None for all until the last one.  This is required for streaming

Reply:
Makes sense. Resolving.